smarty/sampler.py

```diff
@@ -491,32 +529,32 @@ def sample_atomtypes(self):
 
             else:
                 # combinatorial-decorators
-                bondset = [("-","simply"), ("=", "doubly"), ("#","triply"), (":", "aromatic"), ("~","any")]
-                nbondset = len(bondset)
+                nbondset = len(self.bondset)
                 # Pick an atomtype
-                atom1type = self.PickAnAtom(natomtypes)
-                print atom1type
+                #atom1type = self.PickAnAtom(self.atomtypes)
+                atom1type = self.PickAnAtom(self.unmatched_atomtypes)
+                if self.verbose: print("atom1: %s (%s)" % ( atom1type[0], atom1type[1]))
                 # Check if we need to add an alfa or beta substituent
-                if self.HasAlpha(atom1type) == True:
-                    print "has alpha"
+                if self.HasAlpha(atom1type):
```

Comment:
Don't need to check with == True, it's already in boolean form so you can just check what it is.